### PR TITLE
Fix QoS test issues

### PR DESCRIPTION
**Pull request by stephenxs**
### Description of PR

Summary:
Fix QoS test issues on Mellanox platform

Signed-off-by: Stephen Sun <stephens@nvidia.com>

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
Fix QoS test issues on Mellanox platform
- Skip DWRR weight change test on Mellanox platform
- Relax the checking at the beginning of PGSharedWatermark
  It isn't necessary to check the lower bound of the shared buffer pool watermark at the beginning of the test
- Fetch the source and destination ports of the shared headroom pool test from dutConfig instead of using hardcoded numbers

#### How did you do it?
See above

#### How did you verify/test it?
Run QoS test

### CODE REVIEW

File: tests/qos/files/mellanox/qos_param_generator.py [hunk 1]
```diff
@@ -91,8 +92,7 @@ def collect_qos_configurations(self):
                 occupancy_per_port /= 2
                 pkts_num_trig_pfc_shp.append(occupancy_per_port + xon + hysteresis)
                 occupancy_per_port /= 2
-                ingress_ports_list_shp.append(ing_port)
-                ing_port += 1
+                ingress_ports_list_shp.append(testPortIds[i])
```
Comment (neethajohn) [inline tests/qos/files/mellanox/qos_param_generator.py:95]: Need to fix the dst_port_id as well

Comment (stephenxs) [reply tests/qos/files/mellanox/qos_param_generator.py:95]: Not really.
`dst_port_id` is generated by `QosSaiBase`. We don't touch it here.

Comment (neethajohn) [reply tests/qos/files/mellanox/qos_param_generator.py:95]: We overwrite that with the one obtained from qos.yml
https://github.com/Azure/sonic-mgmt/blob/master/tests/qos/test_qos_sai.py#L191

Comment (stephenxs) [reply tests/qos/files/mellanox/qos_param_generator.py:96]: Fixed

**Review by neethajohn — APPROVED** [review]

**Pull request merged**